homeassistant/components/hydrawise.py

```diff
@@ -0,0 +1,156 @@
+            cv.time_period,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Hunter Hydrawise component."""
+    conf = config[DOMAIN]
+    access_token = conf.get(CONF_ACCESS_TOKEN)
+    scan_interval = conf.get(CONF_SCAN_INTERVAL)
+
+    try:
+        from hydrawiser.core import Hydrawiser
+
+        hydrawise = Hydrawiser(user_token=access_token)
+        hass.data[DATA_HYDRAWISE] = HydrawiseHub(hydrawise)
+    except (ConnectTimeout, HTTPError) as ex:
+        _LOGGER.error(
+            "Unable to connect to Hydrawise cloud service: %s", str(ex))
+        hass.components.persistent_notification.create(
+            'Error: {}<br />'
+            'You will need to restart hass after fixing.'
+            ''.format(ex),
+            title=NOTIFICATION_TITLE,
+            notification_id=NOTIFICATION_ID)
+        return False
+
+    def hub_refresh(event_time):
+        """Call Hydrawise hub to refresh information."""
+        _LOGGER.debug("Updating Hydrawise Hub component")
+        hass.data[DATA_HYDRAWISE].data.update_controller_info()
+        dispatcher_send(hass, SIGNAL_UPDATE_HYDRAWISE)
+
+    # Call the Hydrawise API to refresh updates
+    track_time_interval(hass, hub_refresh, scan_interval)
+
+    return True
+
+
+class HydrawiseHub(object):
+    """Representation of a base Hydrawise device."""
+
+    def __init__(self, data):
+        """Initialize the entity."""
+        self.data = data
+
+
+class HydrawiseEntity(Entity):
+    """Entity class for Hydrawise devices."""
+
+    def __init__(self, data, sensor_type):
+        """Initialize the Hydrawise entity."""
+        self.data = data
+        self._sensor_type = sensor_type
+        self._name = "{0} {1}".format(
+            self.data.get('name'),
+            DEVICE_MAP[self._sensor_type][
+                DEVICE_MAP_INDEX.index('KEY_INDEX')])
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        async_dispatcher_connect(
+            self.hass, SIGNAL_UPDATE_HYDRAWISE, self._update_callback)
+
+    @callback
+    def _update_callback(self):
+        """Call update method."""
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def unit_of_measurement(self):
+        """Return the units of measurement."""
+        return DEVICE_MAP[self._sensor_type][
+            DEVICE_MAP_INDEX.index('UNIT_OF_MEASURE_INDEX')]
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        temp = self.hass.data['hydrawise'].data.\
+            controller_info['controllers'][0]['last_contact_readable']
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            'identifier': self.data.get('relay'),
+            'last_contact': temp
```

Comment:
Isn't this time the same as `last_updated` state attribute? If so, please remove this attribute here.

Reply:
`last_contact` is the time the Hydrawise cloud service last connected to the irrigation controller. That's not always the same as `last_updated`.

Reply:
So a hydrawise state can change without contact to the cloud? What about the other way around, have a state always changed after the cloud contacted the controller?

Reply:
I see your point. I removed `last_contact`.